### DISCUSSION TITLE
GitHub Copilot refuses to provide completions with words it deems "sensitive"

### DISCUSSION THREAD

**Discussion by campbellcole** [community/community#110936] [Copilot Conversations]
### Select Topic Area

Product Feedback

### Body

Let me start this by saying I completely understand and agree with the intentions behind this system. I just want to provide a specific case where GitHub Copilot's internal censorship is causing me a lot of trouble in a perfectly valid use case.

I have a project that is built upon a proprietary imaging technique. This technique produces an interleaved streams of videos, one called FTIR and the other called translumination/transillumination.

The problem is that my entire codebase uses the variable name `trans` to represent these transluminated frames. This causes copilot to cut its completions short or flat out refuse to provide any when the completion it would provide contains the word `trans`.

Again, I understand why this was done. I am just voicing my frustration because a significant part of my codebase relies on this terminology, and so my usage of Copilot has naturally become very limited because of this. I do find it odd that this word specifically was included in what seems to be a swear word filter, but of course I'm not the one responsible for the model and what it says.

I would love to see a context-aware sensitive-topic filter. It is a little frustrating that the filtering system is a basic search for bad terms instead of a more sophisticated filtering system, but I completely understand that the added complexity and compute cost could be prohibitive.

Thanks for your time,
Campbell

**Comment by SosthenG** [comment]
I have the same issue with the same workd and it happens a lot because I'm using a translator (Symfony Translator) and the `trans` method is used to translate text. So any generated code that needs to use the translator will not complete (using the IntelliJ/Phpstorm plugin).
See also https://github.com/orgs/community/discussions/7079 seems to be the same issue.
